### ISSUE TITLE
ENOENT: no such file or directory

### ISSUE THREAD

**Issue by Turbo87** [ember-cli/ember-cli#6264]
I'm seeing the following error:

```
The Broccoli Plugin: [Babel] failed with:
Error: ENOENT: no such file or directory, open '/Users/tbn/Code/foobar/tmp/babel-output_path-HghLg9gO.tmp/abc/controllers/something/index.js'
    at Error (native)
    at Object.fs.openSync (fs.js:640:18)
    at Object.fs.readFileSync (fs.js:508:33)
    at Babel.asyncOutputFilteredFile (/Users/tbn/Code/foobar/node_modules/broccoli-persistent-filter/index.js:284:23)
    at lib$rsvp$$internal$$tryCatch (/Users/tbn/Code/foobar/node_modules/rsvp/dist/rsvp.js:1036:16)
    at lib$rsvp$$internal$$invokeCallback (/Users/tbn/Code/foobar/node_modules/rsvp/dist/rsvp.js:1048:17)
    at lib$rsvp$$internal$$publish (/Users/tbn/Code/foobar/node_modules/rsvp/dist/rsvp.js:1019:11)
    at lib$rsvp$asap$$flush (/Users/tbn/Code/foobar/node_modules/rsvp/dist/rsvp.js:1198:9)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)

The broccoli plugin was instantiated at:
    at Babel.Plugin (/Users/tbn/Code/foobar/node_modules/broccoli-plugin/index.js:7:31)
    at Babel.Filter [as constructor] (/Users/tbn/Code/foobar/node_modules/broccoli-persistent-filter/index.js:60:10)
    at new Babel (/Users/tbn/Code/foobar/node_modules/broccoli-babel-transpiler/index.js:35:10)
    at Babel (/Users/tbn/Code/foobar/node_modules/broccoli-babel-transpiler/index.js:30:12)
    at Object.registry.add.toTree (/Users/tbn/Code/foobar/node_modules/ember-cli-babel/index.js:23:52)
    at /Users/tbn/Code/foobar/node_modules/ember-cli-preprocess-registry/preprocessors.js:184:26
    at Array.forEach (native)
    at processPlugins (/Users/tbn/Code/foobar/node_modules/ember-cli-preprocess-registry/preprocessors.js:182:11)
    at module.exports.preprocessJs (/Users/tbn/Code/foobar/node_modules/ember-cli-preprocess-registry/preprocessors.js:175:10)
    at EmberApp.appAndDependencies (/Users/tbn/Code/foobar/node_modules/ember-cli/lib/broccoli/ember-app.js:1072:25)
```

Output from `ember version --verbose && npm --version`:

```
ember-cli: 2.8.0
http_parser: 2.7.0
node: 6.3.0
v8: 5.0.71.52
uv: 1.9.1
zlib: 1.2.8
ares: 1.10.1-DEV
icu: 57.1
modules: 48
openssl: 1.0.2h
os: darwin x64
3.10.3
```
### Steps to reproduce:
1. Create a file with a `SyntaxError`, e.g. like this:
   
   ``` js
   export default Ember.Controller.extend({
    foo
    bar: 'baz',
   });
   ```
2. Start `ember serve`
3. Notice the `SyntaxError`: `The Broccoli Plugin: [Babel] failed with: SyntaxError: foobar/controllers/index.js: Unexpected token (123:2)`
4. Fix the `SyntaxError` by removing the `foo` line
5. Notice the `ENOENT` error

/cc @stefanpenner

**Comment by stefanpenner** [comment]
most likely a a bug in persistent-filter, where it does not reset its internal state when (due to error) its internal stat is in a bad state.

**Comment by Turbo87** [comment]
@stefanpenner https://github.com/stefanpenner/broccoli-persistent-filter/issues/97

**Comment by stefanpenner** [comment]
fixed in 1.3.1 of persistent-filter
